File: 30-days-of-code/day8/Solution.java
import java.util.*;
import java.io.*;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Map<String, String> phoneBook = new HashMap<>();

        for (int i = 0; i < n; i++) {
            String name = sc.next();
            String number = sc.next();
            phoneBook.put(name, number);
        }

        while (sc.hasNext()) {
            String queryName = sc.next();
            if (phoneBook.containsKey(queryName))
                System.out.println(queryName + "=" + phoneBook.get(queryName));
            else
                System.out.println("Not found");
        }
    }
}

File: 30-days-of-code/day8/solution.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int n;
    cin >> n;
    map<string, string> phone_book;

    for (int i = 0; i < n; i++)
    {
        string name, number;
        cin >> name >> number;
        phone_book[name] = number;
    }

    string query_name;
    while (cin >> query_name)
    {
        if (phone_book.count(query_name) > 0)
            cout << query_name << "=" << phone_book[query_name] << "\n";
        else
            cout << "Not found\n";
    }

    return 0;
}